src/components/Dialog/index.tsx

```diff
@@ -101,7 +101,7 @@ const Dialog = ({
   }, [alwaysShowBorder]);
 
   const dialogNode = (
-    <DialogUi.Root defaultOpen>
+    <DialogUi.Root defaultOpen onOpenChange={onDismiss}>
```

Comment:
Add

```
open={isOpen}
``` 

and get rid of `defaultOpen`. We want to make sure the dismiss logic works correctly when we pass in our own state.

Reply:
This fixes the no animation when disablePortal is set to true